File: solutions/python/largest-sum-subarry.py
def find_max_sum_sub_array(A):
  if len(A) < 1:
    return 0

  curr_max = A[0]
  global_max = A[0]
  lengthA = len(A)
  for i in range(1, lengthA):
    if curr_max < 0:
      curr_max = A[i]
    else:
      curr_max += A[i]
    if global_max < curr_max:
      global_max = curr_max
  return global_max

v = [-4, 2, -5, 1, 2, 3, 6, -5, 1]
sum = find_max_sum_sub_array(v)
print("Sum of largest subarray: " + str(sum))

File: solutions/javascript/largest-sum-subarry.js
let findMaxSumSubArray = function(array_) {
  if (array_.length < 1) {
    return 0;
  }

  let currMax = array_[0];
  let globalMax = array_[0];
  let lengtharray = array_.length;
  for (let i = 1; i < lengtharray; i++) {
    if (currMax < 0) {
      currMax = array_[i];
    } else {
      currMax += array_[i];
    }

    if (globalMax < currMax) {
      globalMax = currMax;
    }
  }
  return globalMax;
};

v = [-4, 2, -5, 1, 2, 3, 6, -5, 1];
console.log("Sum of largest subarray: " + findMaxSumSubArray(v));
